rfd/0203-database-healthchecks.md

```diff
@@ -0,0 +1,606 @@
+---
+authors: Gavin Frazar (gavin.frazar@goteleport.com)
+state: draft
+---
+
+# RFD 0203 - Database Health Checks
+
+## Required Approvals
+
+- Engineering: @r0mant && @greedy52
+- Product: @roraback
+
+## What
+
+Teleport database agents will periodically perform health checks to a database's endpoint and report the results as a health status in the agent's `db_server` heartbeat.
```

Comment:
Nit: I would call out that the purpose of this is to make this information visible to users - prioritize benefits to user experience over implementation details.